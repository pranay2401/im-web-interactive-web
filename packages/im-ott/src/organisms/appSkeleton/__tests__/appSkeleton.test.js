import React from "react";
import { cleanup } from "@testing-library/react";
import AppSkeleton from "../AppSkeleton";
import renderer from "react-test-renderer";

afterEach(cleanup);

jest.mock("react-router-dom", () => ({
  Link: "Link",
  useLocation: jest.fn().mockReturnValue({
    pathname: "/search",
    search: "",
    hash: "",
    state: null,
    key: "5nvxpbdafa",
  }),
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

it("AppSkeleton should render correctly", () => {
  const tree = renderer.create(<AppSkeleton />).toJSON();
  expect(tree).toMatchSnapshot();
});
