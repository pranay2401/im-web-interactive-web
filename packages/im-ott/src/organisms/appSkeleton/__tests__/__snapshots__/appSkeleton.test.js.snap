// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSkeleton should render correctly 1`] = `
<section
  className="ant-layout container"
>
  <header
    className="ant-layout-header header webHeader"
  >
    <div
      className="container"
    >
      <Link
        className="logo"
        to="/"
      >
        <span
          aria-label="video-camera"
          className="anticon anticon-video-camera image"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="video-camera"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M912 302.3L784 376V224c0-35.3-28.7-64-64-64H128c-35.3 0-64 28.7-64 64v576c0 35.3 28.7 64 64 64h592c35.3 0 64-28.7 64-64V648l128 73.7c21.3 12.3 48-3.1 48-27.6V330c0-24.6-26.7-40-48-27.7zM328 352c0 4.4-3.6 8-8 8H208c-4.4 0-8-3.6-8-8v-48c0-4.4 3.6-8 8-8h112c4.4 0 8 3.6 8 8v48zm560 273l-104-59.8V458.9L888 399v226z"
            />
          </svg>
        </span>
        <div
          className="text"
        >
          iFlix
        </div>
      </Link>
      <div
        className="browserView"
      >
        <ul
          className="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-light footerMenu"
          data-menu-list={true}
          onKeyDown={[Function]}
          role="menu"
          tabIndex={0}
        >
          <li
            className="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child menuItem"
            data-menu-id={null}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "height": undefined,
                "opacity": 1,
                "order": 0,
                "overflowY": undefined,
                "pointerEvents": undefined,
                "position": undefined,
              }
            }
            tabIndex={-1}
          >
            <span
              aria-label="home"
              className="anticon anticon-home ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="home"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M946.5 505L534.6 93.4a31.93 31.93 0 00-45.2 0L77.5 505c-12 12-18.8 28.3-18.8 45.3 0 35.3 28.7 64 64 64h43.4V908c0 17.7 14.3 32 32 32H448V716h112v224h265.9c17.7 0 32-14.3 32-32V614.3h43.4c17 0 33.3-6.7 45.3-18.8 24.9-25 24.9-65.5-.1-90.5z"
                />
              </svg>
            </span>
            <span
              className="ant-menu-title-content"
            />
          </li>
          <li
            className="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child menuItem"
            data-menu-id={null}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "height": undefined,
                "opacity": 1,
                "order": 1,
                "overflowY": undefined,
                "pointerEvents": undefined,
                "position": undefined,
              }
            }
            tabIndex={-1}
          >
            <span
              aria-label="search"
              className="anticon anticon-search ant-menu-item-icon"
              role="img"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              <svg
                aria-hidden="true"
                data-icon="search"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                />
              </svg>
            </span>
            <span
              className="ant-menu-title-content"
            />
          </li>
          <li
            className="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child menuItem"
            data-menu-id={null}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "height": undefined,
                "opacity": 1,
                "order": 2,
                "overflowY": undefined,
                "pointerEvents": undefined,
                "position": undefined,
              }
            }
            tabIndex={-1}
          >
            <span
              aria-label="heart"
              className="anticon anticon-heart ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="heart"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M923 283.6a260.04 260.04 0 00-56.9-82.8 264.4 264.4 0 00-84-55.5A265.34 265.34 0 00679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 00-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
                />
              </svg>
            </span>
            <span
              className="ant-menu-title-content"
            />
          </li>
          <li
            aria-disabled={true}
            className="ant-menu-overflow-item ant-menu-item ant-menu-item-disabled ant-menu-item-only-child menuItem"
            data-menu-id={null}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            style={
              Object {
                "height": undefined,
                "opacity": 1,
                "order": 3,
                "overflowY": undefined,
                "pointerEvents": undefined,
                "position": undefined,
              }
            }
            tabIndex={null}
          >
            <span
              aria-label="bell"
              className="anticon anticon-bell ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="bell"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M816 768h-24V428c0-141.1-104.3-257.7-240-277.1V112c0-22.1-17.9-40-40-40s-40 17.9-40 40v38.9c-135.7 19.4-240 136-240 277.1v340h-24c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h216c0 61.8 50.2 112 112 112s112-50.2 112-112h216c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM512 888c-26.5 0-48-21.5-48-48h96c0 26.5-21.5 48-48 48zM304 768V428c0-55.6 21.6-107.8 60.9-147.1S456.4 220 512 220c55.6 0 107.8 21.6 147.1 60.9S720 372.4 720 428v340H304z"
                />
              </svg>
            </span>
            <span
              className="ant-menu-title-content"
            />
          </li>
          <li
            className="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child menuItem"
            data-menu-id={null}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "height": undefined,
                "opacity": 1,
                "order": 4,
                "overflowY": undefined,
                "pointerEvents": undefined,
                "position": undefined,
              }
            }
            tabIndex={-1}
          >
            <span
              aria-label="logout"
              className="anticon anticon-logout ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="logout"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M868 732h-70.3c-4.8 0-9.3 2.1-12.3 5.8-7 8.5-14.5 16.7-22.4 24.5a353.84 353.84 0 01-112.7 75.9A352.8 352.8 0 01512.4 866c-47.9 0-94.3-9.4-137.9-27.8a353.84 353.84 0 01-112.7-75.9 353.28 353.28 0 01-76-112.5C167.3 606.2 158 559.9 158 512s9.4-94.2 27.8-137.8c17.8-42.1 43.4-80 76-112.5s70.5-58.1 112.7-75.9c43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 7.9 7.9 15.3 16.1 22.4 24.5 3 3.7 7.6 5.8 12.3 5.8H868c6.3 0 10.2-7 6.7-12.3C798 160.5 663.8 81.6 511.3 82 271.7 82.6 79.6 277.1 82 516.4 84.4 751.9 276.2 942 512.4 942c152.1 0 285.7-78.8 362.3-197.7 3.4-5.3-.4-12.3-6.7-12.3zm88.9-226.3L815 393.7c-5.3-4.2-13-.4-13 6.3v76H488c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h314v76c0 6.7 7.8 10.5 13 6.3l141.9-112a8 8 0 000-12.6z"
                />
              </svg>
            </span>
            <span
              className="ant-menu-title-content"
            />
          </li>
          <li
            aria-hidden={true}
            className="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="none"
            style={
              Object {
                "height": 0,
                "opacity": 0,
                "order": 9007199254740991,
                "overflowY": "hidden",
                "pointerEvents": "none",
                "position": "absolute",
              }
            }
          >
            <div
              aria-controls={null}
              aria-expanded={false}
              aria-haspopup={true}
              className="ant-menu-submenu-title"
              data-menu-id={null}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              role="menuitem"
              style={null}
              tabIndex={-1}
              title={null}
            >
              <span
                aria-label="ellipsis"
                className="anticon anticon-ellipsis"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="ellipsis"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                  />
                </svg>
              </span>
              <i
                className="ant-menu-submenu-arrow"
              />
            </div>
          </li>
        </ul>
        <div
          aria-hidden={true}
          style={
            Object {
              "display": "none",
            }
          }
        />
      </div>
    </div>
  </header>
  <main
    className="ant-layout-content content"
  />
  <footer
    className="ant-layout-footer footer"
  >
    <div
      className="webFooter"
    >
      <div
        className="footerContainer"
      >
        <div
          className="footerWrapper webWrapper"
        >
          <div
            className="socialContainer"
          >
            <Link
              className="social"
              to="#"
            >
              <span
                aria-label="facebook"
                className="anticon anticon-facebook"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="facebook"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-92.4 233.5h-63.9c-50.1 0-59.8 23.8-59.8 58.8v77.1h119.6l-15.6 120.7h-104V912H539.2V602.2H434.9V481.4h104.3v-89c0-103.3 63.1-159.6 155.3-159.6 44.2 0 82.1 3.3 93.2 4.8v107.9z"
                  />
                </svg>
              </span>
            </Link>
            <Link
              className="social"
              to="#"
            >
              <span
                aria-label="twitter"
                className="anticon anticon-twitter"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="twitter"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M928 254.3c-30.6 13.2-63.9 22.7-98.2 26.4a170.1 170.1 0 0075-94 336.64 336.64 0 01-108.2 41.2A170.1 170.1 0 00672 174c-94.5 0-170.5 76.6-170.5 170.6 0 13.2 1.6 26.4 4.2 39.1-141.5-7.4-267.7-75-351.6-178.5a169.32 169.32 0 00-23.2 86.1c0 59.2 30.1 111.4 76 142.1a172 172 0 01-77.1-21.7v2.1c0 82.9 58.6 151.6 136.7 167.4a180.6 180.6 0 01-44.9 5.8c-11.1 0-21.6-1.1-32.2-2.6C211 652 273.9 701.1 348.8 702.7c-58.6 45.9-132 72.9-211.7 72.9-14.3 0-27.5-.5-41.2-2.1C171.5 822 261.2 850 357.8 850 671.4 850 843 590.2 843 364.7c0-7.4 0-14.8-.5-22.2 33.2-24.3 62.3-54.4 85.5-88.2z"
                  />
                </svg>
              </span>
            </Link>
            <Link
              className="social"
              to="#"
            >
              <span
                aria-label="instagram"
                className="anticon anticon-instagram"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="instagram"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 306.9c-113.5 0-205.1 91.6-205.1 205.1S398.5 717.1 512 717.1 717.1 625.5 717.1 512 625.5 306.9 512 306.9zm0 338.4c-73.4 0-133.3-59.9-133.3-133.3S438.6 378.7 512 378.7 645.3 438.6 645.3 512 585.4 645.3 512 645.3zm213.5-394.6c-26.5 0-47.9 21.4-47.9 47.9s21.4 47.9 47.9 47.9 47.9-21.3 47.9-47.9a47.84 47.84 0 00-47.9-47.9zM911.8 512c0-55.2.5-109.9-2.6-165-3.1-64-17.7-120.8-64.5-167.6-46.9-46.9-103.6-61.4-167.6-64.5-55.2-3.1-109.9-2.6-165-2.6-55.2 0-109.9-.5-165 2.6-64 3.1-120.8 17.7-167.6 64.5C132.6 226.3 118.1 283 115 347c-3.1 55.2-2.6 109.9-2.6 165s-.5 109.9 2.6 165c3.1 64 17.7 120.8 64.5 167.6 46.9 46.9 103.6 61.4 167.6 64.5 55.2 3.1 109.9 2.6 165 2.6 55.2 0 109.9.5 165-2.6 64-3.1 120.8-17.7 167.6-64.5 46.9-46.9 61.4-103.6 64.5-167.6 3.2-55.1 2.6-109.8 2.6-165zm-88 235.8c-7.3 18.2-16.1 31.8-30.2 45.8-14.1 14.1-27.6 22.9-45.8 30.2C695.2 844.7 570.3 840 512 840c-58.3 0-183.3 4.7-235.9-16.1-18.2-7.3-31.8-16.1-45.8-30.2-14.1-14.1-22.9-27.6-30.2-45.8C179.3 695.2 184 570.3 184 512c0-58.3-4.7-183.3 16.1-235.9 7.3-18.2 16.1-31.8 30.2-45.8s27.6-22.9 45.8-30.2C328.7 179.3 453.7 184 512 184s183.3-4.7 235.9 16.1c18.2 7.3 31.8 16.1 45.8 30.2 14.1 14.1 22.9 27.6 30.2 45.8C844.7 328.7 840 453.7 840 512c0 58.3 4.7 183.2-16.2 235.8z"
                  />
                </svg>
              </span>
            </Link>
          </div>
          <div
            className="footerLogo"
          >
            © 2021 iflix Media. All rights reserved.
          </div>
        </div>
      </div>
    </div>
  </footer>
</section>
`;
